apps/wallet/src/ui/app/redux/slices/transaction-requests/index.ts

```diff
@@ -65,10 +66,105 @@ export const loadTransactionResponseMetadata = createAsyncThunk<
     }
 );
 
+export type TxnMetaResponse = {
+    coinSymbol?: string | null;
+    amount?: number | null;
+    objectId?: string;
+    isListing?: boolean;
+} | null;
+
+const getRequestCost = (
+    txEffects: TransactionEffects,
+    address: string
+): TxnMetaResponse | null => {
+    const events = txEffects?.events || [];
+
+    let sumCoin = 0;
+    const coinMeta: {
```

Comment:
Shouldn't this really me a multitude of these? Any transaction could have multiples of these, right?